### PR TITLE
add support for node templates in workerconfig

### LINKED ISSUE #580: Allow configuration of NodeTemplate for Autoscaler by user
**How to categorize this issue?**

/area auto-scaling
/area usability
/kind enhancement
/platform openstack

**What would you like to be added**:

Currently the nodeTemplate is filled only on the basis of cloudprofile. There is a requirement to able to configure and update the nodeTemplate through shoot Yaml.

**Why is this needed**:
Scalemp memory one could enhance memory at runtime, so autoscaler could be made aware of it using nodeTemplate.

Very similar to https://github.com/gardener/gardener-extension-provider-azure/pull/519

### PR DESCRIPTION
**Pull request by kon-angelo**
**How to categorize this PR?**

/area control-plane
/kind enhancement
/platform openstack

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:
Fixes #580 

**Special notes for your reviewer**:

**Release note**:

```other operator
Add support for user-provided `NodeTemplate`s
```

### CODE REVIEW

**Review by MartinWeindel — DISMISSED** [review]
/lgtm

**Review by timebertt — COMMENTED** [review]
Interesting change. I'm missing documentation that explains the motivation for when user would want to use this feature and a few examples :)

**Review by himanshu-kun — DISMISSED** [review]
/lgtm
could you also update the end user docs https://github.com/gardener/gardener-extension-provider-openstack/blob/master/docs/usage-as-end-user.md#workerconfig

**Comment by kon-angelo** [comment]
@timebertt this is to support ScaleMP for openstack. The details are similar to these comment: https://github.com/gardener/gardener/issues/4948#issuecomment-997815132 and allows the user to override the expected machine specs

**Comment by kon-angelo** [comment]
Or if your comment was more about that we need to update the user docs for this change, I will follow up with a new commit asap :)

**Comment by kon-angelo** [comment]
@himanshu-kun @timebertt I added the docs. Let me know if this is sufficient.

**Review by himanshu-kun — DISMISSED** [review]
/lgtm

File: docs/usage-as-end-user.md [hunk 1]
```diff
@@ -137,15 +137,23 @@ apiVersion: openstack.provider.extensions.gardener.cloud/v1alpha1
...
 
+### ServerGroups
 When you specify the `serverGroup` section in your worker group configuration, a new server group will be created with the configured policy for each worker group that enabled this setting and all machines managed by this worker group will be assigned as members of the created server group.
 
 For users to have access to the server group feature, it must be enabled on the `CloudProfile` by your operator. 
 Existing clusters can take advantage of this feature by updating the server group configuration of their respective worker groups. Worker groups that are already configured with server groups can update their setting to change the policy used, or remove it altogether at any time.
 
 Users must be aware that **any change to the server group settings will result in a rolling deployment of new nodes for the affected worker group**.
 
+### Node Templates
```
Comment (timebertt) [inline docs/usage-as-end-user.md:155]: Section as such looks good, but should probably be placed after the next paragraph (which is still on the topic of server groups)

Comment (kon-angelo) [reply docs/usage-as-end-user.md]: fixed. That was silly of me

**Pull request merged**